tests/components/sleepiq/test_binary_sensor.py

```diff
@@ -1,67 +1,48 @@
 """The tests for SleepIQ binary sensor platform."""
-import unittest
-
 import requests_mock
 
 from homeassistant.components.sleepiq import binary_sensor as sleepiq
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
 
 from tests.async_mock import MagicMock
-from tests.common import get_test_home_assistant
 from tests.components.sleepiq.test_init import mock_responses
 
+CONFIG = {"username": "foo", "password": "bar"}
 
-class TestSleepIQBinarySensorSetup(unittest.TestCase):
-    """Tests the SleepIQ Binary Sensor platform."""
-
-    DEVICES = []
-
-    def add_entities(self, devices):
-        """Mock add devices."""
-        for device in devices:
-            self.DEVICES.append(device)
 
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.username = "foo"
-        self.password = "bar"
-        self.config = {"username": self.username, "password": self.password}
-        self.DEVICES = []
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    @requests_mock.Mocker()
-    def test_setup(self, mock):
-        """Test for successfully setting up the SleepIQ platform."""
+async def test_sensor_setup(hass):
+    """Test for successfully setting up the SleepIQ platform."""
+    with requests_mock.Mocker() as mock:
```

Comment:
This is available as a fixture that you can hook into your tests:

```suggestion
async def test_sensor_setup(hass, requests_mock):
    """Test for successfully setting up the SleepIQ platform."""
```